### PR TITLE
Fixes #517

### LINKED ISSUE #517: save and download?
At the moment, if you make a change to your code and then click "download" what downloads is the latest save and not the latest code. I think that download should probably trigger an "auto-save" so that it's really "save and download". . .thoughts?

Also, it might be useful to name the zip file with a timestamp, i.e. `project_yr_mm_dd_time` if a project is downloaded multiple times it allows for some baby steps into version control?

### PR DESCRIPTION
**Pull request by shinytang6**
Before your pull request is reviewed and merged, make sure you

* [x] there are no linting errors -- `npm run lint`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master. If you're asked to make more changes make sure you rebase onto master then too!
* [x] pull request is descriptively named and links to an issue number, i.e. `Fixes #123`

Thank you! Fixes #517

### CODE REVIEW

**Comment by shinytang6** [comment]
@catarak  please review:)

File: server/controllers/project.controller.js [hunk 1]
```diff
@@ -261,7 +261,39 @@ function buildZip(project, req, res) {
     res.status(500).send({ error: err.message });
   });
 
-  res.attachment(`${project.name}.zip`);
+  function currentTime() {
```
Comment (catarak) [inline server/controllers/project.controller.js]: included in this project is a library called [moment](https://momentjs.com/) which makes it easy to convert times to a formatted string. i would prefer if you used this library as it converts this function to one or two lines of code.

**Comment by shinytang6** [comment]
Done. l didn't notice that library:)

**Comment by catarak** [comment]
even though this idea wasn't in the issue, i think it would also make sense as part of this PR to slugify (an npm module which is already included in the project now) the sketch name, since it's annoying to have spaces in file names, see https://github.com/processing/p5.js-web-editor/blob/master/server/models/project.js#L43 for an example

**Comment by shinytang6** [comment]
thanks for reminding this!

**Comment by catarak** [comment]
this is working great!

**Pull request merged**